### ISSUE TITLE
[Event Request] - OnAfterSetFALedgEntryFiltersInCalcFAPostedOriginalAcqCostAmount Codeunit 5626 "FA General Report"

### ISSUE THREAD

**Issue by oldmat** [microsoft/ALAppExtensions#21505]
Hello, please consider adding the following event to Codeunit 5626 "FA General Report"

```
   [IntegrationEvent(false, false)]
    local procedure OnAfterSetFALedgEntryFiltersInCalcFAPostedOriginalAcqCostAmount(var FALedgEntry: Record "FA Ledger Entry"; Period: Option "Before Starting Date","Net Change","at Ending Date"; StartingDate: Date; EndingDate: Date)
    begin
    end;
```

and call it accordingly inside the "CalcFAPostedOriginalAcqCostAmount" procedure:

```
    procedure CalcFAPostedOriginalAcqCostAmount(FANo: Code[20]; Period: Option "Before Starting Date","Net Change","at Ending Date"; StartingDate: Date; EndingDate: Date; DeprBookCode: Code[10]): Decimal
    var
        FALedgEntry: Record "FA Ledger Entry";
    begin
        with FALedgEntry do begin
            if EndingDate = 0D then
                EndingDate := 99991231D;
            SetRange("FA Posting Category", "FA Posting Category"::" ");
            SetRange("FA No.", FANo);
            SetRange("Depreciation Book Code", DeprBookCode);
            SetRange("Part of Book Value", true);
            SetRange("FA Posting Type", "FA Posting Type"::"Acquisition Cost");
            case Period of
                Period::"Before Starting Date":
                    SetRange("FA Posting Date", 0D, StartingDate - 1);
                Period::"Net Change":
                    SetRange("FA Posting Date", StartingDate, EndingDate);
                Period::"at Ending Date":
                    SetRange("FA Posting Date", 0D, EndingDate);
            end;
            SetRange("Reclassification Entry", true);
            if FindFirst() then
                SetRange("Entry No.", 0, "Entry No." - 1);
            SetRange("Reclassification Entry");

            **OnAfterSetFALedgEntryFiltersInCalcFAPostedOriginalAcqCostAmount(FALedgEntry, Period, StartingDate, EndingDate);**

            CalcSums(Amount);
            exit(Amount);
        end;
    end;
```

Thanks!

**Comment by JesperSchulz** [comment]
Thanks for reporting this. We agree, and we’ll publish a fix asap, either in an update for the current version or in the next major release. Please do not reply to this, as we do not monitor closed issues. If you have follow-up questions or requests, please create a new issue where you reference this one.

Build ID: 53382.
